src/main/java/bio/terra/service/filedata/flight/ingest/IngestBuildAndWriteScratchLoadFileStep.java

```diff
@@ -71,22 +77,15 @@ public StepResult doSkippableStep(FlightContext context) {
           jsonNodes
               .peek(
                   node -> {
-                    for (var columnName : fileColumns) {
+                    for (var column : fileColumns) {
+                      String columnName = column.getName();
                       JsonNode fileRefNode = node.get(columnName);
                       if (fileRefNode != null && fileRefNode.isObject()) {
-                        // replace
-                        BulkLoadFileModel fileModel =
-                            Objects.requireNonNull(
-                                objectMapper.convertValue(fileRefNode, BulkLoadFileModel.class));
-                        String fileKey = fileModel.getTargetPath();
-                        if (pathToFileIdMap.containsKey(fileKey)) {
-                          String fileId = pathToFileIdMap.get(fileKey);
-                          ((ObjectNode) node).put(columnName, fileId);
-                        } else {
-                          // If the file wasn't ingest, clear the node.
-                          ((ObjectNode) node).removeAll();
-                          failedRowCount.getAndIncrement();
-                        }
+                        replaceJsonObject(
+                            node, fileRefNode, columnName, pathToFileIdMap, failedRowCount);
+                      } else if (fileRefNode != null && fileRefNode.isArray()) {
```

Comment:
Instead of testing `fileRefNode != null` twice, I'd check it once and use `continue` above this `if`.